src/components/MapboxMap.tsx

```diff
@@ -0,0 +1,157 @@
+import { useRef, useEffect } from 'react';
+import mapboxgl, { Map, Marker, MapboxOptions, LngLatBounds, MarkerOptions, LngLat } from 'mapbox-gl';
+import { Result, useSearchState } from '@yext/search-headless-react';
+import { useDebouncedFunction } from '../hooks/useDebouncedFunction';
+import ReactDOM from 'react-dom';
+
+/**
+ * A functional component that can be used to render a custom marker on the map.
+ *
+ * @public
+ */
+export type PinComponent<T> = (props: { result?: Result<T>, index?: number }) => JSX.Element;
+
+/**
+ * A function use to derive a result's coordinate.
+ *
+ * @public
+ */
+export type CoordinateGetter<T> = (result: Result<T>) => Coordinate | undefined;
+
+/**
+ * Coordinate use to represent the result's location on a map.
+ *
+ * @public
+ */
+export interface Coordinate {
+  /** The latitude of the location. */
+  latitude: number,
+  /** The longitude of the location. */
+  longitude: number
+}
+
+/**
+ * Props for the {@link MapboxMap} component.
+ *
+ * @public
+ */
+export interface MapboxMapProps<T> {
+  /** Mapbox access token. */
+  mapboxAccessToken: string,
+  /** Interface for customizing the {@link MapboxMap} map derived from Mapbox API's Map options. */
+  mapboxOptions?: Omit<MapboxOptions, 'container'>,
```

Comment:
is styling customizable through these options?

Reply:
are you referring to the style like `style: 'mapbox://styles/mapbox/streets-v11'`? If so, yup that field is included in `MapboxOptions` interface

Reply:
the type for `style` is: ` style?: mapboxgl.Style | string | undefined;`. User can either select one of the predefined styles (https://docs.mapbox.com/api/maps/styles/) or make their own following the interface of `mapboxgl.Style`:
<img width="300" alt="Screen Shot 2022-09-26 at 4 42 01 PM" src="https://user-images.githubusercontent.com/36055303/192375939-6748e219-214d-4a57-adf3-61fa7f7467bc.png">